packages/pluggableWidgets/datagrid-date-filter-web/src/components/FilterComponent.tsx

```diff
@@ -22,12 +24,21 @@ interface FilterComponentProps {
     screenReaderInputCaption?: string;
     tabIndex?: number;
     styles?: CSSProperties;
-    updateFilters?: (value: Date | null, type: DefaultFilterEnum) => void;
+    updateFilters?: (
+        value: Date | null,
+        rangeValues: [Date | null, Date | null] | null,
+        type: DefaultFilterEnum
+    ) => void;
 }
 
 export function FilterComponent(props: FilterComponentProps): ReactElement {
     const [type, setType] = useState<DefaultFilterEnum>(props.defaultFilter);
     const [value, setValue] = useState<Date | null>(null);
+    const [rangeValues, setRangeValues] = useState<[Date | null, Date | null] | null>(
+        props.defaultStartDate || props.defaultEndDate
+            ? [props.defaultStartDate ?? null, props.defaultEndDate ?? null]
```

Comment:
Question: [in the DataGridFilter updateFilter callback we do a lot of `??` and then turning it back to `undefined` instead of using null](https://github.com/mendix/widgets-resources/pull/1271/files?diff=unified&w=0#diff-b61ac2888d18582bbb952c95c26da608aa5e5222ad4affa9076d749e961ba7efR127). Can we not make it `undefined` here and then avoid all of that?

Reply:
Fixed